crates/starknet-devnet-core/src/state/mod.rs

```diff
@@ -284,11 +284,19 @@ impl CustomState for StarknetState {
         let compiled_class = contract_class.clone().try_into()?;
 
         if let ContractClass::Cairo1(cairo_lang_contract_class) = &contract_class {
-            let cairo_lang_compiled_class = cairo_lang_starknet_classes::casm_contract_class::CasmContractClass::from_contract_class(
-                cairo_lang_contract_class.clone(), true, MAX_BYTECODE_SIZE_LIMIT).map_err(|_| Error::SierraCompilationError
-            )?;
-            let casm_hash =
-                Felt::new(cairo_lang_compiled_class.compiled_class_hash().to_be_bytes())?;
+            let casm_json = usc::compile_contract(
+                serde_json::to_value(cairo_lang_contract_class)
+                    .map_err(|err| Error::SerializationError { origin: err.to_string() })?,
+            )
+            .map_err(|_| Error::SierraCompilationError)?;
+
+            let casm_hash = Felt::from(
+                serde_json::from_value::<CompiledClass>(casm_json)
+                    .map_err(|err| Error::DeserializationError { origin: err.to_string() })?
+                    .class_hash()
+                    .map_err(|err| Error::UnexpectedInternalError { msg: err.to_string() })?,
+            );
```

Comment:
Perhaps extract to a `casm_json_to_hash` function.

Reply:
[b05f918](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/391/commits/b05f9187a30724d91fd9b78d16c59fc53a9ea601)